source/slang/slang-ir-byte-address-legalize.cpp

```diff
@@ -617,6 +619,66 @@ struct ByteAddressBufferLegalizationContext
             }
         }
 
+        if (m_options.lowerBasicTypeOps)
+        {
+            // Some platforms e.g. Metal does not allow loading basic types that are not 4-byte sized.
+            // We need to lower such loads.
+            IRSizeAndAlignment sizeAlignment;
+            SLANG_RETURN_NULL_ON_FAIL(getNaturalSizeAndAlignment(m_targetProgram->getOptionSet(), type, &sizeAlignment));
+            if (sizeAlignment.size == 8)
+            {
+                // We need to load the value as two 4-byte values and then combine them.
+                auto loOffset = offset;
+                auto hiOffset = emitOffsetAddIfNeeded(offset, 4);
+                IRInst* loadLoArgs[] = { buffer, loOffset };
+                IRInst* loadHiArgs[] = { buffer, hiOffset };
+                auto loLoad = m_builder.emitIntrinsicInst(m_builder.getUIntType(), kIROp_ByteAddressBufferLoad, 2, loadLoArgs);
+                auto hiLoad = m_builder.emitIntrinsicInst(m_builder.getUIntType(), kIROp_ByteAddressBufferLoad, 2, loadHiArgs);
```

Comment:
Do we need to worry about endian-ness?
I guess we can go with little-endian and fix if there are any problems later.

Reply:
Eventually we should but all targets we care about are little endian so we can live with the looseness here for now, I think.